src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                            F.If(basePrintCall,
+                                F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(", ")))));
+                    }
+                }
+
+                Debug.Assert(!printableMembers.IsEmpty && block is object);
+
+                for (var i = 0; i < printableMembers.Length; i++)
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(member.Name))));
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(" = "))));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
+                        F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendObject),
+                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+
+                    if (i < printableMembers.Length - 1)
+                    {
+                        block.Add(
+                            F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(", "))));
+                    }
+                }
+
+                block.Add(F.Return(F.Literal(true)));
+
+                F.CloseMethod(F.Block(block.ToImmutableAndFree()));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                diagnostics.Add(ex.Diagnostic);
+                F.CloseMethod(F.ThrowNull());
+            }
+        }
+
+        internal static MethodSymbol? FindValidPrintMethod(TypeSymbol containingType, CSharpCompilation compilation)
+        {
+            if (containingType.IsErrorType() || containingType.IsObjectType())
+            {
+                return null;
+            }
+
+            MethodSymbol? candidate = null;
+            var accessibility = (containingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && containingType.IsSealed) ?
```

Comment:
nit: it is easier to read the conditional expression if the `?` `:` token is at the beginning of the line, not the end of the line.